src/Filters/filt_order.jl

```diff
@@ -0,0 +1,458 @@
+
+[4] P. K. Mogensen and A. N. Riseth, Optim: A mathematical optimization package for Julia. Journal of Open 
+Source Software, 2018. https://github.com/JuliaNLSolvers/Optim.jl/blob/master/src/univariate/solvers/brent.jl
+
+====================================================================#
+
+
+
+toprototype(Wp::Real, Ws::Real, ftype::Type{Lowpass}) = Ws / Wp
+toprototype(Wp::Real, Ws::Real, ftype::Type{Highpass}) = Wp / Ws
+function toprototype(Wp::Tuple{Real,Real}, Ws::Tuple{Real,Real}, ftype::Type{Bandpass}) 
+    # bandpass filter must have two corner frequencies we're computing with
+    Wa = (Ws.^2 .- Wp[1]*Wp[2]) ./ (Ws .* (Wp[1]-Wp[2]))
+    minimum(abs.(Wa))
+end
+toprototype(Wp::Tuple{Real, Real}, Ws::Tuple{Real, Real}, Rp::Real, Rs::Real, ftype::Type{Bandstop}) = butterworth_bsfmin(Wp, Ws, Rp, Rs)
+fromprototype(Wp::Real, Wscale::Real, ftype::Type{Lowpass}) = Wp * Wscale
+fromprototype(Wp::Real, Wscale::Real, ftype::Type{Highpass}) = Wp / Wscale
+
+function fromprototype(Wp::Tuple{Real,Real}, Wscale::Real, ftype::Type{Bandstop})
+    Wa = zeros(2)
+    diff = Wp[2]-Wp[1]
+    prod = Wp[2]*Wp[1]
+    Wa[1] = (diff + sqrt(diff^2 + 4*(Wscale^2)*prod)) / (2*Wscale)
+    Wa[2] = (diff - sqrt(diff^2 + 4*(Wscale^2)*prod)) / (2*Wscale)
+    sort!(abs.(Wa))
+end
+
+function fromprototype(Wp::Tuple{Real,Real}, Wscale::Real, ftype::Type{Bandpass})
+    Wsc = [-Wscale, Wscale]
+    Wa = -Wsc .* (Wp[2]-Wp[1])./2 .+ sqrt.( Wsc.^2/4*(Wp[2]-Wp[1]).^2 .+ Wp[1]*Wp[2])
+    sort!(abs.(Wa))
+end
+
+butterworth_order_estimate(Rp::Real, Rs::Real, warp::Real) = (log(db2pow(Rs) - 1) - log(db2pow(Rp) - 1)) / (2*log(warp))
+natfreq_estimate(warp::Real, Rs::Real, order::Integer) = warp / (db2pow(Rs) - 1)^(1/(2*order))
+
+function elliptic_order_estimate(Rp::Real, Rs::Real, Wa::Real)
+    # Elliptic integer order estmate. Requires Complete Elliptic integral of first kind.
+    ϵ = √(db2pow(Rp) - 1) # define selectivity/discrimination parameters.
+    k₁ = ϵ / √(db2pow(Rs) - 1)
+    k = Wa^-1
+    (k^2 < 1) || throw(DomainError(k^2, "Selectivity parameter specifies too narrow of a transition width.")) # check if in-bounds (k₁ << k <~ 1)
+    (1-k₁^2 < 1) || throw(DomainError(1-k₁^2, "Discrimination parameter specifies too deep of a stopband."))
+    K = tuple(ellipk(k^2), ellipk(1-k^2)) # define the complementary moduli
+    K₁ = tuple(ellipk(k₁^2), ellipk(1-k₁^2))
+    
+    # other order approach would be using (k₁'/k₁) / (k′/k).
+    (K[1]*K₁[2])/(K[2]*K₁[1])
+end
+
+function chebyshev_order_estimate(Rp::Real, Rs::Real, Wa::Real)
+    # Chebyshev1/2 order estimate. Requires inverse hyperbolic cosine.
+    ϵs, ϵp = db2pow(Rs) - 1, db2pow(Rp) - 1    
+    acosh(√(ϵs / ϵp)) / acosh(Wa) # Eq. (10.3.63) in [1]
+end
+
+function brent(f, x1::T, x2::T) where T <: AbstractFloat
+    a, b = x1, x2
+    fa, fb = f(a), f(b)
+    c = copy(a) # midpoint of [b, a] bracket
```

Comment:
`c` seems unused before the next assignment below also is not accessed outside the `while` loop, so what's this line for? Can we just get rid of it?
```suggestion
```